### DISCUSSION TITLE
Put passphrase remember disclaimer about safety?

### DISCUSSION THREAD

**Discussion by robigan** [tejado/obsidian-gpgCrypt#16] [General]
GPG Agent as far as I'm aware goes to great lengths to protect its memory space from memory scanning processes to extract keys. JavaScript doesn't have great memory safety features, and I doubt Obsidian's JS engine does either. Users who might be paranoid about security should know that by storing the passphrase, it can theoretically be possible to extract the passphrase from memory.

LoC for change
https://github.com/tejado/obsidian-gpgCrypt/blob/003aef79d3f2f9079a0ca27172e2adaee9763949/src/settings/SettingsTab.ts#L80-L98

And of course I'd be willing to PR this myself

**Comment by tejado** [comment]
Users that are paranoid about such things should do a proper risk analysis and might then not use Obsidian in the first place.
If someone is able to extract the password from memory, the corresonding "attacker" could probably also install a hardware keylogger or adjust the OS settings to hook into the configured gpg passphrase dialog.

Advanced users that are aware of this are likely use smartcards which is only supported over the gpg wrapper mode.

To avoid adding such "insecure" disclaimers all over the place (e.g. for the notes itself), I have no plans to add this.

**Reply by robigan** [reply]
Fair enough, I came to a similar conclusion, but as I see it, people are _uneducated_, and in the spirit of educating people, I would put a disclaimer about this to tell them about the risks of such. You and I know about these risks, but my father who might be searching for a way to hide secrets using a markdown editor, having just learned about OpenPGP and GPG, might do this without realizing the implications of such.

**Reply by tejado** [reply]
Educating people with a disclaimer is very hard. Users would always read it like "OpenPGP.js is unsafe/less safe" without really understanding what that means and what the rsisks are. So this leads to the case that most users want to use the "safe" version and try to switch to gpg CLI Wrapper. There is no guidance to install it so they are on their own (configuration, key generation, ...) and probably will fail on it and just uninstall gpgpCrypt.

In the case that they will be successful, they did not win anything as the notes are also readable over memory. So I would need to put a disclaimer there as well without any solution (the only one is to not use Obsidian for such cases). 
Also I would then need to "educate" about hardware keyloggers and all other risks, which is not the scope of this plugin.

And your example with your father: why should he be aware of the implications? 
Is he using it for work? Then he wouldn't be allowed anyway.
Is he using it personally? Than its perfect to use. Probably his wife will not extract the passphrase from memory.

We can also look at other (much more popular) projects like mailvelope. They also don't have such disclaimer and use OpenPGP.js as the default without mentioning any implications or saying that it is unsafe.
